src/Reflection/ReflectionParameter.php

```diff
@@ -208,8 +209,20 @@ private function parseDefaultValueNode() : void
         $defaultValueNode = $this->node->default;
 
         if ($defaultValueNode instanceof Node\Expr\ClassConstFetch) {
+            $className = $this->getRealClassName(implode('\\', $defaultValueNode->class->parts));
+
+            if ($className instanceof Self_) {
+                $class = $this->function->getDeclaringClass();
+                do {
+                    if ($class->hasConstant($defaultValueNode->name)) {
+                        $className = $class->getName();
+                        break;
```

Comment:
This `do/while` should be in a private method, with this particular statement being a `return`